### DISCUSSION TITLE
Proposal: Sub-minute interval schedules

### DISCUSSION THREAD

**Discussion by davidmorch** [PrefectHQ/prefect#4120] [Ideas] [answered]
Hey,

I have been using Prefect for around half a year now and I am looking into the possibility of using Prefect as our only scheduler. Currently we're using Prefect and or own custom scheduler. The only reason we're using our own scheduler is due to the lack of support for sub-minute schedules.

This could be a pretty basic thing from a user perspective, but I know there is a lot behind the scenes that needs to be implemented for this to work properly. The syntax's in Prefect could look as:
```python
schedule = IntervalSchedule(
    start_date=pendulum.datetime(2021, 2, 14, 10, 0, 0),
    interval=pendulum.duration(seconds=20)
)
```

I have a lot of different use cases that my team and I use Prefect for; ETL, business monitoring, data collection, high computation flows, automation & machine learning.  Where we of course want to collect everything in one place because of maintenance and simplicity. 

**So my specific use case is quite simply:**
I have a data source that is updated quite frequently and therefore want to retrieve the data from this source every 20 second.
The flow's schedule could just look as above where you then add an interval duration with 20 seconds.

I would like to hear if this is something other people would like to see as a feature and if the Prefect team has been discussing this before.

Thanks for your time and consideration!

**Comment by cicdw** [comment, marked as answer]
Hi @davidmorch - allowing for sub-minute auto-scheduling will make it difficult to ensure no runs are ever missed; because of this risk vs. the marginal reward, this is very low on our priority list to consider.  That being said, this is not so hard to achieve already by using ad-hoc runs instead of the Prefect scheduler - there is no interval limit for ad-hoc runs and they can be created to start in the future using `scheduled_start_time`, so you could create these runs by calling the `create_flow_run` GraphQL route from a client.  You could either implement this as a trigger on your data change (to avoid the need to schedule in the future), or you could schedule out the runs every day or so.

**Reply by davidmorch** [reply]
Hi Chris, 
Thank you for taking your time and answering and explaining! 
I knew there was a reason behind since you guys didn't already implemented this, but I wanted it to give it a go and hear you guys out.  Thank you for giving me an alternative solution 👍
